jasminewd2/index.d.ts

```diff
@@ -44,6 +44,13 @@ declare namespace jasmine {
     toThrowError(expected?: new (...args: any[]) => Error | Promise<new (...args: any[]) => Error>, message?: string | RegExp | Promise<string | RegExp>): Promise<void>;
   }
 
+  interface ArrayLikeMatchers<T extends ArrayLike<T>> extends Matchers<T> {
```

Comment:
@lukas-zech-software ```Array``` vs ```ArrayLike``` doesn't matter, that's not the issue. We are suggesting that you drop the ```T extends ArrayLike<T>``` constraint. This is what you would end up with:

```ts
interface ArrayLikeMatchers<T> extends Matchers<ArrayLike<T>> {
 toBe(expected: Expected<ArrayLike<T>>, expectationFailOutput?: any): Promise<void>;
 toEqual(expected: Expected<ArrayLike<T>>, expectationFailOutput?: any): Promise<void>;
 toContain(expected: T, expectationFailOutput?: any): Promise<void>;
 not: ArrayLikeMatchers<T>;
}
```

If you disagree that this would work, could you demonstrate your use case with a test case? Perhaps you're trying to do something subtle here.